packages/desktop/components/popups/send/SendConfirmationPopup2.svelte

```diff
@@ -0,0 +1,235 @@
+        giftStorageDeposit,
+        visibleSurplus,
+        isInternal,
+        layer2Parameters
+    )
+
+    async function rebuildTransactionOutput(): Promise<void> {
+        updateNewTransactionDetails({
+            type: transactionType,
+            expirationDate,
+            giftStorageDeposit,
+        })
+
+        try {
+            const transactionDetails = get(newTransactionDetails)
+            const outputParams = await getOutputParameters(transactionDetails)
+            preparedOutput = await prepareOutput($selectedAccount.index, outputParams, getDefaultTransactionOptions())
+
+            await updateStorageDeposit()
+
+            if (transactionDetails.expirationDate === undefined) {
+                initialExpirationDate = getInitialExpirationDate(expirationDate, storageDeposit, giftStorageDeposit)
+            }
+        } catch (err) {
+            handleError(err)
+        }
+    }
+
+    async function updateStorageDeposit(): Promise<void> {
+        const { storageDeposit: _storageDeposit, giftedStorageDeposit: _giftedStorageDeposit } =
+            await getStorageDepositFromOutput($selectedAccount, preparedOutput as CommonOutput)
+
+        storageDeposit = minimumStorageDeposit = _storageDeposit > 0 ? _storageDeposit : _giftedStorageDeposit
+
+        if (isBaseTokenTransfer) {
+            const rawAmount = Number((transactionDetails as NewTokenTransactionDetails).rawAmount)
+            if (rawAmount >= storageDeposit) {
+                storageDeposit = 0
+            }
+        }
+
+        // Note: we need to adjust the surplus
+        // so we make sure that the surplus is always added on top of the minimum storage deposit
+        if (Number(surplus) > 0) {
+            if (minimumStorageDeposit >= Number(surplus)) {
+                visibleSurplus = surplus = undefined
+            } else {
+                visibleSurplus = Number(surplus) - minimumStorageDeposit
+                // Note: we have to hide it because currently, in the sdk,
+                // the storage deposit return strategy is only looked at
+                // if the provided amount is < the minimum required storage deposit
+                hideGiftToggle = true
+            }
+        }
+    }
+
+    function toggleGiftStorageDeposit(): void {
+        giftStorageDeposit = !giftStorageDeposit
+    }
+
+    async function sendOutputAndClosePopup(): Promise<void> {
+        await sendOutput(preparedOutput)
+        closePopup()
+    }
+
+    async function onConfirmClick(): Promise<void> {
+        try {
+            validateSendConfirmation(preparedOutput as CommonOutput)
+
+            if ($isActiveLedgerProfile) {
+                ledgerPreparedOutput.set(preparedOutput)
+            }
+
+            updatePopupProps({ isSendAndClosePopup: true, preparedOutput, calculatedStorageDeposit: storageDeposit })
+            await checkActiveProfileAuth(sendOutputAndClosePopup, { stronghold: true, ledger: false })
```

Comment:
When returning from unlocking stronghold popup:

-  Transaction is not automatically sent but now another click to Send is required
- Expiration Date is also removed

1. before clicking send
![image](https://github.com/iotaledger/firefly/assets/52735957/56895f61-59de-4043-b02a-1b7a2f02329e)
2. after returning from unlocking stronghold popup
![image](https://github.com/iotaledger/firefly/assets/52735957/6d6cae3e-fe7e-444e-8cba-2fa9ecf9dd68)